File: images-basics/pixelAccessing.cpp
//-----------------------------------------------------------------------------
//
// Example for pixel accessing in C++.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

#include <iostream>

int main() {
    cv::Mat img = cv::imread("../img.jpg", cv::IMREAD_COLOR);

    // Access to (0, 0) coordinates pixel.
    cv::Scalar intensity1 = img.at<uchar>(0, 0);

    std::cout << "(0, 0) - " << intensity1 << '\n';

    // Another way.
    cv::Scalar intensity2 = img.at<uchar>(cv::Point(0, 0));

    std::cout << "(100, 50) - " << intensity2 << '\n';

    cv::Vec3b intensity3 = img.at<cv::Vec3b>(0, 0);

    std::cout << "\nB - " << static_cast<int>(intensity3.val[0]) << '\n';
    std::cout << "R - " << static_cast<int>(intensity3.val[1]) << '\n';
    std::cout << "G - " << static_cast<int>(intensity3.val[2]) << '\n';

    // And pixel changing.
    img.at<uchar>(0, 0) = 128;
}

File: images-basics/pixelAccessing.py
#-----------------------------------------------------------------------------
#
# Example for pixel accessing in Python.
#
#-----------------------------------------------------------------------------

import cv2
import numpy as np

img = cv2.imread("../img.jpg", cv2.IMREAD_COLOR)

# Access to (0, 0) coordinates pixel.
intensity1 = img[0, 0]

print("(0, 0) -", intensity1)

# Another way.
intensity2 = {img.item(0), img.item(1), img.item(2)}

print("(100, 50) -", intensity2)

intensity3 = img[0, 0]

print("\nB -", intensity3[0])
print("G -", intensity3[1])
print("R -", intensity3[2])

# And pixel changing.
img[0, 0] = 128
